sqlconnect/internal/base/tableadmin.go

```diff
@@ -132,6 +132,15 @@ func (db *DB) TableExists(ctx context.Context, relation sqlconnect.RelationRef)
 
 // ListColumns returns a list of columns for the given table
 func (db *DB) ListColumns(ctx context.Context, relation sqlconnect.RelationRef) ([]sqlconnect.ColumnRef, error) {
+	// check if relation exists before returning columns
+	exists, err := db.TableExists(ctx, relation)
+	if err != nil {
+		return nil, err
+	}
+	if !exists {
+		return nil, fmt.Errorf("cannot fetch columns for %s: relation does not exist", relation.String())
+	}
+
```

Comment:
Instead of performing this second query, it should be safe to assume that if there are no columns found, the relation doesn't exist.

Reply:
updated